### ISSUE TITLE
Creating a note from a template via wikilink doesn't work if the template has a custom filepath

### ISSUE THREAD

**Issue by JosNun** [foambubble/foam#1327]
### Describe the bug

When using the tooltip to create a note from a template, and the template has a `filepath` specified, the wikilink is not linked up. I assume this is because the filepath no longer matches the text in the wikilink.

### Steps to Reproduce the Bug or Issue

- Create a template that changes the filepath of notes created with it
- Create a wikilink, and use the hover tooltip "Create note from template" option to create the note
- Note is created with the correct filename, but the wikilink is not updated with the right name

Example Template:
```md
---
foam_template:
  name: 'Example Template'
  description: 'An example for reproducing a bug'
  filepath: '$FOAM_DATE_DATE-$FOAM_SLUG.md'
---
```

### Expected behavior

The text of the wikilink should be updated to match the new notes title. 
For example:
- Given a wikilink `[[test]]`, and a template that creates a file named `2024-01-26-test.md`
- I would expect that the wikilink gets updated to either `[[2024-01-26-test]]`, `(test)[2024-01-26-test]` (probably not accurate, since it creates a markdown link instead of a wikilink), or create a link reference definitioon at the bottom of the page

### Screenshots or Videos

https://github.com/foambubble/foam/assets/7729878/5828f8ba-8a21-4b02-8efc-6201016bbfb5

### Operating System Version

macOS 14.3

### Visual Studio Code Version

1.85.2 (Universal)

**Comment by riccardoferretti** [comment]
Thanks for the clear bug report.
Yes, this definitely needs to be addressed, the expected behavior is that the wikilink will point to the new note using the appropriate Foam Identifier.

**Comment by JosNun** [comment]
For posterity sake, a good spot to start reverse-engineering this would be taking a look at where it's implemented https://github.com/foambubble/foam/pull/1061/files

**Comment by riccardoferretti** [comment]
Fixed in 0.25.9
